### PR TITLE
Add optional 3d range calculation mode to warheads

### PR DESCRIPTION
**Pull request by jrb0001**

### CODE REVIEW

**Review by CDVoidwalker — APPROVED** [review]
Looks good to me apart of the naming, but I honestly cannot think of better names without changing those that are already in

**Comment by GraionDilach** [comment]
I find it very unappealing that I haven't been consulted or notified of this PR nor heard any of the reasoning behind it nor were asked to review it even.

@jrb0001 if you intend to keep up with your private patches with the above still applied, you'll hit a stage when I will drop CD support altogether.

**Comment by jrb0001** [comment]
@GraionDilach I have asked you on discord but there was no reponse from you yet. I am not able to add you (nor Voidwalker) as a reviewer of the PR for some reason. This is more a proof of concept than something to be merged immediately so please give your feedback.

The idea is to remove the need to create two separate weapons to avoid that double damage issue if air and ground targets are in range.

File: OpenRA.Game/WorldUtils.cs [hunk 1]
```diff
@@ -43,6 +43,13 @@ public static IEnumerable<Actor> FindActorsInCircle(this World world, WPos origi
 				a => (a.CenterPosition - origin).HorizontalLengthSquared <= r.LengthSquared);
 		}
 
+		public static IEnumerable<Actor> FindActorsInSphere(this World world, WPos origin, WDist r)
+		{
+			var vec = new WVec(r, r, r);
+			return world.ActorMap.ActorsInBox(origin - vec, origin + vec).Where(
```
Comment (GraionDilach) [inline OpenRA.Game/WorldUtils.cs:49]: This is definitely wrong. ActorsInBox will result with a box.

Comment (GraionDilach) [reply OpenRA.Game/WorldUtils.cs:49]: Ah, this is what FindActorsInCircle is doing... niec.

**Review by CDVoidwalker — APPROVED** [review]
Looks good to me

**Review by GraionDilach — APPROVED** [review]
LGTM but keep all changes local for now. I consider this a bugfix and I don't see why there would be a need for the boolean even.

**Comment by jrb0001** [comment]
It seems to be unneeded anyway (at least for Bullet with SpreadDamge). @CDVoidwalker, if something else has the problem, it would be know the exact combination to create a dedicated testcase for it.

**Pull request closed without merging**